### PR TITLE
Rename row -> dataObject, rows -> data

### PR DESCRIPTION
**Pull request by DanTup**
To avoid confusion with table rows, the data is internally referred to as "data" (the whole list) and "dataObject" (a single object from that list of data).

The method for setting externally is still called setRows() since externally the table rows are not exposed and this feels logical, but we can always change it to setData if we'd like total consistency.

### CODE REVIEW

**Comment by DanTup** [comment]
Turns out `data` was used in one place externally (logging) but I hadn't saved the file. So maybe it's a bit weird we call `setRows` but then do `table.data.length`. LMK if you think we should rename `setRows` -> `setData` (or if you have better names). This is mainly to avoid confusion around what "rows" are in the table virtualisation code (which is somewhat complex).

File: lib/logging/logging.dart [hunk 1]
```diff
@@ -92,7 +92,7 @@ class LoggingScreen extends Screen {
   }
 
   void _updateStatus() {
-    final int count = loggingTable.rows.length;
+    final int count = loggingTable.data.length;
```
Comment (devoncarew) [inline lib/logging/logging.dart]: Perhaps expose a `rowCount` getter on `Table`?

Comment (DanTup) [reply lib/logging/logging.dart]: Good call :)

Done, and also moved the constructors to fix the lint. Running "Sort Members" on these files makes a LOT of changes - we may want to do it at some point when there aren't too many PRs open!

**Comment by DanTup** [comment]
> does it sort in Flutter order?

Oh, no - doesn't seem to. It does put constructors before methods, but doesn't put them first. I think one of the codebases I contribute to normally sorts (analysis server?) but I think I'm useless at remembering, so my code normally gets sorted by others!

**Comment by devoncarew** [comment]
Yes, analysis server sorts, but most everything else leaves things where they are (and Flutter just wants ctors first).

**Pull request merged**